src/index.js

```diff
@@ -31,11 +31,31 @@ class DynamicOverflow extends React.Component {
 
   calculateSize = () => {
     const containerWidth = this.containerNode && this.containerNode.clientWidth;
-    const firstChildWidth = this.tabNode && this.tabNode.clientWidth;
 
-    const maximumChildrenAllowed =
-      Math.floor(containerWidth / firstChildWidth) - 1;
-    const currentChildrenCount = this.props.list({}).length;
+    let moreNodeWidth = 0;
+    if (this.moreNode) {
+      const moreNodeStyle = window.getComputedStyle(this.moreNode);
+        const moreNodeMarginWidth = parseInt(moreNodeStyle.marginLeft) + parseInt(moreNodeStyle.marginRight);
+        moreNodeWidth = this.moreNode.offsetWidth + moreNodeMarginWidth;
+    }
+
+    let maximumChildrenAllowed = 0;
+    let nodeWidthSum = 0;
+
+    this.listNodes.filter(node => node !== null).forEach(
+      (node) => {
+        let nodeStyle = window.getComputedStyle(node);
+        let nodeWidth = node.offsetWidth;
+        let nodeMarginWidth = parseInt(nodeStyle.marginLeft) + parseInt(nodeStyle.marginRight);
+
+        nodeWidthSum += (nodeWidth + nodeMarginWidth);
+        if (nodeWidthSum + moreNodeWidth < containerWidth) {
+          maximumChildrenAllowed += 1;
+        }
```

Comment:
Small optimization: we should `break` out of the loop if the node elements are larger than the container.